pkg/operator/controller/ingress/deployment.go

```diff
@@ -309,7 +309,16 @@ func desiredRouterDeployment(ci *operatorv1.IngressController, ingressController
 		// because we need the rule on multi-node clusters, even if
 		// desiredReplicas is 1, so that a rolling update schedules
 		// the new pod to the same node as the old pod.
+		pointerTo := func(ios intstr.IntOrString) *intstr.IntOrString { return &ios }
```

Comment:
We should change this to `ptr.To[intstr.IntOrString]`, but not in this PR as that would complicate backports and as we have several places where we should make this change.